addons/point_of_sale/static/src/app/store/product_configurator_popup/product_configurator_popup.js

```diff
@@ -105,22 +106,26 @@ export class ProductConfiguratorPopup extends AbstractAwaitablePopup {
     }
 
     getPayload() {
-        const selected_attributes = [];
+        const attribute_custom_values = [];
         let attribute_value_ids = [];
         var price_extra = 0.0;
         const quantity = this.state.quantity;
 
         this.env.attribute_components.forEach((attribute_component) => {
-            const { value, valueIds, extra } = attribute_component.getValue();
-            selected_attributes.push(value);
+            const { valueIds, extra, custom_value } = attribute_component.getValue();
             attribute_value_ids.push(valueIds);
+
+            if (custom_value) {
+                attribute_custom_values[valueIds] = custom_value;
```

Comment:
there could be something wrong here because "valueIds" seems to be an array of ids.